src/Aspire.Dashboard/Components/Pages/Resources.razor.cs

```diff
@@ -818,4 +825,22 @@ private async Task ContextMenuClosed(Microsoft.AspNetCore.Components.Web.MouseEv
         _contextMenuClosedTcs?.TrySetResult();
         _contextMenuClosedTcs = null;
     }
+
+    // IComponentWithTelemetry impl
+    public ComponentTelemetryContext TelemetryContext { get; } = new(DashboardUrls.ResourcesBasePath);
+
+    public void UpdateTelemetryProperties()
+    {
+        var properties = new List<ComponentTelemetryProperty>
+        {
+            new(TelemetryPropertyKeys.ResourceView, new AspireTelemetryProperty(PageViewModel.SelectedViewKind.ToString(), AspireTelemetryPropertyType.UserSetting))
+        };
+
+        foreach (var resourceTypeGroup in _resourceByName.Values.GroupBy(r => r.ResourceType))
+        {
+            properties.Add(new ComponentTelemetryProperty($"{TelemetryPropertyKeys.ResourceType}.{resourceTypeGroup.Key}", new AspireTelemetryProperty(resourceTypeGroup.Count(), AspireTelemetryPropertyType.Metric)));
```

Comment:
Resource type is being used in the key. What happens if the resource type contains dots? Or anything else odd or unusual. Does it need to be escaped?.